File: command/Tv.java
package command;

public class Tv {
    private String name;

    public Tv(String name) {
        this.name = name;
    }

    public void turnOn() {
        System.out.println("Turning on TV");
    }

    public void turnOff() {
        System.out.println("Turning off TV");
    }

    public String getName() {
        return name;
    }
}

File: typescript/patterns/command/tv.ts
class Tv {
  private name: string;

  constructor(name: string) {
    this.name = name;
  }

  turnOn(): void {
    console.log(`Turning on TV ${this.name}...`);
  }

  turnOff(): void {
    console.log(`Turning off TV ${this.name}...`);
  }

  getName(): string {
    return this.name;
  }
}
